File: backend/src/main/java/com/bomnuocv1/domain/entity/RecentTransaction.java
package com.bomnuocv1.domain.entity;

import lombok.Builder;
import lombok.Getter;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

@Getter
@Builder
public class RecentTransaction {

    private final UUID id;
    private final String type; // PUMP or PAYMENT
    private final String farmerName;
    private final String details; // e.g. "Bơm 2.5 công • 08:30 sáng" or "Thu nợ cũ • Hôm qua"
    private final BigDecimal amount; // positive for pump (+500.000), negative for payment (-1.200.000) or vice versa
    private final String formattedAmount; // e.g. "+ 500.000đ" or "- 1.200.000đ"
    private final String statusBadge; // "Ghi nợ" or "Đã thu"
    private final Instant createdAt;
}

File: frontend/app/src/main/java/com/vn/bomnuocv1/domain/model/RecentTransaction.kt
package com.vn.bomnuocv1.domain.model

import java.math.BigDecimal

data class RecentTransaction(
    val id: String,
    val type: String, // "PUMP" or "PAYMENT"
    val farmerName: String,
    val details: String,
    val amount: BigDecimal,
    val formattedAmount: String,
    val statusBadge: String,
    val createdAt: String?
)
